src/v/storage/tests/storage_e2e_test.cc

```diff
@@ -2624,8 +2625,23 @@ FIXTURE_TEST(write_truncate_compact, storage_test_fixture) {
     // Ensure we've cleaned up all our staging segments such that a removal of
     // the log results in nothing leftover.
     auto dir_path = log.config().work_directory();
-    mgr.remove(ntp).get();
-    BOOST_REQUIRE_EQUAL(false, ss::file_exists(dir_path).get());
+    try {
+        mgr.remove(ntp).get();
```

Comment:
I had an itchy merge finger on this one, but now I'm not so sure: wasn't the test failure in the assertion, rather than being a throw from the `.remove` call?

Now we're swallowing exceptions from `.remove` but not doing the file_exists check at all.

Reply:
`remove()` throws if its call to `remove_file(ntp_dir)` fails due to it being non-empty. The `file_exists` was more of a sanity check -- if `remove()` succeeds it should always be empty.

Note this isn't a boost assertion:
```
unknown location(0): fatal error: in "write_truncate_compact": std::__1::__fs::filesystem::filesystem_error: filesystem error: remove failed: Directory not empty ["test.data.Udl24iEXrc/default/test/0_0"]
```

Reply:
ah, ok.

Looking again at https://github.com/redpanda-data/redpanda/pull/7912, it seems like the directory walk in `log_manager::remove` could also log any unexpected files (while only deleting the unexpected files that end with ".staging"., that way the test wouldn't have to, and we'd have the logs in any non-unit-test situations where it happened.